e2e/testcases/gatekeeper_test.go

```diff
@@ -30,6 +30,10 @@ func TestConstraintTemplateAndConstraintInSameCommit(t *testing.T) {
 	// TODO enable the test on autopilot clusters when GKE 1.21.3-gke.900 reaches regular/stable.
 	nt := nomostest.New(t, nomostesting.Reconciliation1, ntopts.Unstructured, ntopts.SkipAutopilotCluster)
 
+	crdName := "k8sallowedrepos.constraints.gatekeeper.sh"
+	nt.T.Logf("Delete the %q CRD if needed", crdName)
+	nt.MustKubectl("delete", "crd", crdName, "--ignore-not-found")
```

Comment:
Seems like this logic belongs in the `ApplyGatekeeperCRD` function

Reply:
The `ApplyGatekeeperCRD` function applies a given CRD. 

There are two CRDs involved in this e2e test: `constrainttemplates.templates.gatekeeper.sh` and `k8sallowedrepos.constraints.gatekeeper.sh`.

This e2e test works fine if the `constrainttemplates.templates.gatekeeper.sh`  CRD already exists before the test starts.

This e2e test fails if the `k8sallowedrepos.constraints.gatekeeper.sh` CRD exists before the test starts. 

Therefore, we need to make sure the `k8sallowedrepos.constraints.gatekeeper.sh` CRD does not exist before the test starts.